File: IntegerBreak3.cpp
#include<iostream>
#include<math.h>
#include<vector>

using namespace std;

class Solution {
public:
    int integerBreak(int n) {
        vector<int> dp{0, 1, 1, 2, 4, 6, 9};
        for(int i = 7; i <= n; i++) {
        	dp.push_back(3 * dp[i - 3]);
		}
		
		return dp[n];
		
    }
};

int main() {
	
	int n;
	cin>>n;
	
	Solution *solution = new Solution();
	cout<<solution->integerBreak(n);
	
	return 0;
	
}

File: IntegerBreak3.py
class Solution:
    def integerBreak(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 3:
            return n - 1
        if n == 4:
            return n
        n -= 5
        return int(pow(3, n // 3 + 1) * (n % 3 + 2))


if __name__ == '__main__':
    n = int(input())
    solution = Solution()
    print(solution.integerBreak(n))
